src/Tempest/Http/src/Route.php

```diff
@@ -30,9 +30,12 @@ public function __construct(
     ) {
         // Routes can have parameters in the form of "/{PARAM}/",
         // these parameters are replaced with a regex matching group
-        $matchingRegex = preg_replace(
-            '#\{(\w+)}#',
-            '([^/]++)',
+        $matchingRegex = preg_replace_callback(
+            '#\{(\w+)(?::([^}]+))?\}#',
```

Comment:
Something I find helpful for clarity is to use named capture groups: `(?<name>?::([^}]+))`

Then you can access it via `$matches['name']`